proposed/http-handlers/request-handlers.md

```diff
@@ -110,20 +110,18 @@ use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 
 /**
+ * Participant in processing a server request and response
+ *
  * An HTTP middleware component participates in processing an HTTP message,
- * either by acting on the request or the response. This interface defines the
- * methods required to use the middleware.
+ * either by acting on the request or the response. The middleware may create a
+ * response if conditions prevent the request from being processed further.
  */
 interface MiddlewareInterface
 {
     /**
-     * Process an incoming server request and return a response, optionally delegating
-     * response creation to a handler.
-     *
-     * @param ServerRequestInterface $request
-     * @param RequestHandlerInterface $handler
+     * Processes a request and the resulting response
      *
-     * @return ResponseInterface
+     * May delegate creating the response to the request handler.
      */
```

Comment:
I suggest the following:

```php
/**
 * Process an incoming server request
 *
 * Processes an incoming server request in order to produce a response.
 * If unable to produce the response itself, it may delegate to the provided
 * request handler to do so.
 */
```

Reply:
I don't think I agree with this wording. It implies that the primary purpose of the middleware is to produce a response, when that is actually the purpose of the request handler.

Reply:
> It implies that the primary purpose of the middleware is to produce a response

It is. It's not incompatible with the fact that a request handler can also do that.

Reply:
Fair enough.